suro-kafka-producer/src/main/java/com/netflix/suro/sink/kafka/KafkaSink.java

```diff
@@ -105,34 +119,155 @@ private void setServoReporter() {
         props.put(ServoReporter.class.getName(), ServoReporter.class);
     }
 
+    private AtomicLong queuedRecords = new AtomicLong(0);
+    private AtomicLong sentRecords = new AtomicLong(0);
+    private AtomicLong droppedRecords = new AtomicLong(0);
+    private volatile Action3 recordCounterListener;
+    public void setRecordCounterListener(Action3 action) {
+        this.recordCounterListener = action;
+    }
+
+    private ConcurrentSkipListSet<String> metadataFetchedTopicSet = new ConcurrentSkipListSet<>();
+    private PublishSubject<MessageContainer> stream = PublishSubject.create();
+    private Subscription subscription;
+    private ExecutorService executor = Executors.newSingleThreadExecutor(
+        new ThreadFactoryBuilder().setDaemon(false).setNameFormat("kafka-metadata-fetcher-%d").build());
+
     @Override
-    public void writeTo(MessageContainer message) {
+    public void writeTo(final MessageContainer message) {
+        queuedRecords.incrementAndGet();
+        runRecordCounterListener();
+
+        if (metadataFetchedTopicSet.contains(message.getRoutingKey())) {
```

Comment:
now I think about it. does this mean we can't ever use auto-topic-creation? this is probably not an issue for suro sink via LogManager.logEvent.